sklearn/pipeline.py

```diff
@@ -145,6 +145,9 @@ def set_params(self, **kwargs):
         return self
 
     def _validate_steps(self):
+        if isinstance(self.steps, tuple):
+            self.steps = list(self.steps)
```

Comment:
what do you suggest?

it is modified anyway in : L226

`self.steps[step_idx] = (name, fitted_transformer)``

which is what created the problem